### PR TITLE
Align cell content

### PR DESCRIPTION
**Pull request by tjconcept**
Make the output more pretty.

```sh
$ node cli.js < fixtures/input.json --align right
|          name | repo                                                                | desc                                                                                        |
| ------------: | ------------------------------------------------------------------- | ------------------------------------------------------------------------------------------- |
|       trilogy | [citycide/trilogy](https://github.com/citycide/trilogy)             | No-hassle SQLite with type-casting schema models and support for native & pure JS backends. |
|         strat | [citycide/strat](https://github.com/citycide/strat)                 | Functional-ish JavaScript string formatting, with inspirations from Python.                 |
| tablemark-cli | [citycide/tablemark-cli](https://github.com/citycide/tablemark-cli) | Generate markdown tables from JSON data at the command line.                                |
```

And pasted to GitHub:

|          name | repo                                                                | desc                                                                                        |
| ------------: | ------------------------------------------------------------------- | ------------------------------------------------------------------------------------------- |
|       trilogy | [citycide/trilogy](https://github.com/citycide/trilogy)             | No-hassle SQLite with type-casting schema models and support for native & pure JS backends. |
|         strat | [citycide/strat](https://github.com/citycide/strat)                 | Functional-ish JavaScript string formatting, with inspirations from Python.                 |
| tablemark-cli | [citycide/tablemark-cli](https://github.com/citycide/tablemark-cli) | Generate markdown tables from JSON data at the command line.                                |

I've done some pretty extensive changes and I have no clue if this is a desirable direction to you.

### CODE REVIEW

**Comment by citycide** [comment]
Hello again @tjconcept! I like this but we need to maintain compatibility down to node 4. That means no `String#pad{Start,End}` or `String#repeat` and other features that were added in node 8. It would probably be fine to add these as small pure helper functions rather than a polyfill.

With that fixed up I think I'd merge this prior to https://github.com/citycide/tablemark-cli/pull/3 so this also hits the cli.

File: index.js [hunk 1]
```diff
@@ -19,48 +16,92 @@ module.exports = (input, options) => {
...
+  )
+
+  let alignments = keys.map((key, i) => {
+    if (Array.isArray(options.columns)
+      && options.columns[i]
+      && options.columns[i].align
+    ) {
+      let align = String(options.columns[i].align).toUpperCase()
+
+      if (!ALIGN.includes(align)) throw new TypeError(
```
Comment (citycide) [inline index.js]: Let's use curly braces here.

```js
if (!ALIGN.includes(align)) {
  throw new TypeError(`Unknown alignment, got ${options.columns[i].align}`)
}
```

File: index.js [hunk 2]
```diff
@@ -19,48 +16,92 @@ module.exports = (input, options) => {
...
+        typeof item[key] === 'undefined' ? 0 : String(item[key]).length,
+        sizes[i]
+      )
+    ),
+    titles.map(t => t.length)
+  )
+
+  let alignments = keys.map((key, i) => {
+    if (Array.isArray(options.columns)
+      && options.columns[i]
```
Comment (citycide) [inline index.js]: Probably would be good to pull up that `options.columns[i]` value into a variable if we're using it multiple times like this.

File: index.js (hunk 1 above)
Comment (tjconcept) [reply index.js]: This is a difference from my usual style, and though I tried to be consistent with yours, this one slipped.

**Comment by tjconcept** [comment]
Out of curiosity, why keep support down to 4.x?

**Comment by citycide** [comment]
> Out of curiosity, why keep support down to 4.x?

I end up having to use 4.x here and there, and it's still in support mode until April '18. Can't wait to drop it but that's the situation. :laughing:

File: index.js (hunk 1 above)
Comment (citycide) [reply index.js]: All good, it's a nitpick. I wouldn't mind if it weren't for the `throw` spanning multiple lines.

File: index.js (hunk 2 above)
Comment (tjconcept) [reply index.js]: Which style would you prefer then? Nested `if`s or "return early"?

**Comment by tjconcept** [comment]
I also have a one-line change to uppercase titles derived from keys, if it's welcome, I'll open a PR once this is merged.

File: index.js (hunk 2 above)
Comment (citycide) [reply index.js]: I go for returning early, but I wouldn't say the code you're replacing exemplified that at all :man_facepalming: I'm actually fine with this the way you've got it.

Comment (tjconcept) [reply index.js]: I usually prefer "return early", and started this code path as such, but it's twice the lines, so I rolled back to this.

File: index.js (hunk 1 above)
Comment (citycide) [reply index.js]: Ah I forgot about `Array#includes` - for node 4 support we'll have to use the ol' `indexOf`. I think that's the last thing here.

File: index.js (hunk 2 above)
Comment (citycide) [reply index.js]: I'm good with that. No point in being rigid just for the sake of 'style'. This looks fine :+1:

**Comment by citycide** [comment]
Looks great, thanks!

**Pull request merged**